modules/developer_manual/pages/testing/unit-testing.adoc

```diff
@@ -73,20 +74,91 @@ You can find more information in {phpunit-docs-url}[the PHPUnit documentation].
 
 === Running PHP Unit tests for ownCloud >= 10.0
 
-There are existing tests provided by ownCloud which are ready to run.
+There are existing test options provided by ownCloud.
+To run them, change into the root directory of your ownCloud installation and run `grep "make test" <(make help)` to see tests and parameters available.
 
-* Change into `webroot` and run `make help` to see tests and parameters
-available.
+You should see output similar to the below example.
 
-Testing apps
+[source,console]
+----
+make test                   run all tests
+make test-php-unit          run all PHP tests
+make test-php-style         run PHP code style checks
+make test-php-phan          run PHP phan static code analyzer
+make test-php-phpstan       run PHP phpstan static code analyzer
+make test-js                run Javascript tests
+make test-js-debug          run Javascript tests in debug mode (continuous)
+make test-acceptance-api    run API acceptance tests
+make test-acceptance-cli    run CLI acceptance tests
+make test-acceptance-webui  run webUI acceptance tests
+make test-php-unit          TEST_DATABASE=mysql TEST_PHP_SUITE=path/to/testfile.php
+make test-php-style-fix     run PHP code style checks and fix any issues found
+----
+
+==== Testing Apps
+
+To run the tests for a specific app with the provided PHPUnit version:
 
-* To run the tests for a specific app with the provided PHPUnit version,
-change into `<webroot>/apps/<appname>` and
+. Change into one of the writable directories listed in the `apps_paths` array in `config/config.php`. For example:
+[source,console]
+----
+cd apps-external
+----
 
+. Clone the app from GitHub. For example:
+[source,console]
+----
+git clone https://github.com/owncloud/notes.git
+----
+
+. Enable the app. For example:
+[source,console]
```

Comment:
You must use `[source,console,subs="attributes+"]` as written in my comment before,
else the {...} replacement below will not work.

Reply:
I have no idea about all this "magic" - I just write code and automated tests :)
Adding this now.